### ISSUE TITLE
[JENKINS-12875] "No valid crumb was included in the request" errors all around

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#2433]
<p>After upgrading Jenkins from 1.448 to 1.451 (with yum, upgraded plugins too) several parts of Jenkins have become unusable with error "No valid crumb was included in the request":</p>

<h4><a name="Consoleoutput"></a>Console output</h4>

<ul>
	<li>When building, the log is unavailable, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21553/21553_console_log.png" title="console_log.png attached to JENKINS-12875">console_log.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
	<li>Once the build is done, the complete log is available, but the same error appears on the side <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21552/21552_console_side.png" title="console_side.png attached to JENKINS-12875">console_side.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="Buildconfiguration"></a>Build configuration</h4>

<ul>
	<li>Builds cannot be launched anymore from the build page (The firebug error console displays the "No valid crumb was included in the request" error). They still can be launched from the dashboard though.</li>
	<li>Cannot add build steps (when choosing a build step type an error is reported in background), see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21556/21556_add_build_step.png" title="add_build_step.png attached to JENKINS-12875">add_build_step.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="ManageJenkinsconfiguration"></a>Manage Jenkins configuration</h4>

<ul>
	<li>The page loads only partially (first half is permanently greyed out, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21555/21555_configuration.png" title="configuration.png attached to JENKINS-12875">configuration.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span>). The firebug error console displays the "No valid crumb was included in the request" error.</li>
</ul>

<h4><a name="ManageJenkinsplugins"></a>Manage Jenkins plugins</h4>

<ul>
	<li>Cannot upgrade nor downgrade, see error shown <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21554/21554_plugins.png" title="plugins.png attached to JENKINS-12875">plugins.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

---
<details><summary><i>Originally reported by <a title="Núria Aloy" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=naloy">naloy</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-12875" target="_blank">"No valid crumb was included in the request" errors all around</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Daniel Beck" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-04-01T23:57:26+00:00
<li><b>imported</b>: 2025-11-04
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=naloy">naloy</a>:</i>
<p>After yet another restart things seems to be back to normal (???!!)</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=valentine">valentine</a>:</i>
<p>Not completely normal. The loading thing in the management section is still happening </p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=naloy">naloy</a>:</i>
<p>In the Jenkins configuration page:</p>

<p>Error: Form.findMatchingInput is not a function<br/>
Source File: <a href="http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js</a><br/>
Line: 338</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=windyroad">windyroad</a>:</i>
<p>I'm not as blocked as the reporting user, but I'm having the same problem (no valid crumb..). In my case it only seems to affect the console output, though the error can be seen for "ajaxBuildQueue" and "ajaxExecuters" on the config page, the home page (and I'm assuming others).</p>

<p>What's interesting is that I only get this problem through a reverse proxy. <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a> works fine, but <a href="http://ci.windyroad.org/" class="external-link" target="_blank" rel="nofollow noopener">http://ci.windyroad.org/</a> (which is RPed to  <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a>) does not.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=windyroad">windyroad</a>:</i>
<p>I've offered  USD100.00 at FreedomSponsors to have this fixed (<a href="http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around" class="external-link" target="_blank" rel="nofollow noopener">http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around</a>).<br/>
Once you solve it (according to the acceptance criteria described there), just create a FreedomSponsors account and mark it as resolved (oh, you'll need a Paypal account too)<br/>
I'll then check it out and will gladly pay.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=onemanbucket">onemanbucket</a>:</i>
<p>Pull request: <a href="https://github.com/jenkinsci/jenkins/pull/596" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/596</a></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=algob">algob</a>:</i>
<p>I've same issue, if I try to disable a project (use Jenkins v1.502):<br/>
   <a href="http://rhodan:8080/jenkins/job/myproject_3562/disable" class="external-link" target="_blank" rel="nofollow noopener">http://rhodan:8080/jenkins/job/myproject_3562/disable</a></p>

<p>It looks like, I would have the same issue, if I would try to enable it, if it is disabled: <a href="https://issues.jenkins.io/browse/JENKINS-14501" title="Enable Job link fails with "No valid crumb was included in the request"" class="issue-link" data-issue-key="JENKINS-14501"><del>JENKINS-14501</del></a></p>

<p>I tried to added these line to /opt/tomcat/bin/catalina.sh, but it didn't solve the issue:<br/>
   &#8212; SNIP &#8212;<br/>
JAVA_OPTS="$JAVA_OPTS -Dhudson.security.csrf.requestfield=crumb"<br/>
   &#8212; SNAP &#8212;</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=daemoen">daemoen</a>:</i>
<p>Further testing of this shows that even with the working setups above, chrome still does not like it.  Firefox works fine.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=davea_br">davea_br</a>:</i>
<p>I can confirm that this also occurs on Centos 6.5 with Nginx 1.6 using Firefox. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=elephantjim">elephantjim</a>:</i>
<p>I'm using Jenkins behind a reverse proxy using nginx, too.  I tried to set ignore_invalid_headers to "off" on my server's nginx settings to no avail.  But then I found this: <a href="http://trac.nginx.org/nginx/ticket/117" class="external-link" target="_blank" rel="nofollow noopener">http://trac.nginx.org/nginx/ticket/117</a></p>

<p>It turns out that nginx needs to process headers before selecting a virtual server: the "Host" header determines which server to use.  Unless the ignore_invalid_headers is set to off on the default server, it won't get read.  The other way is to set it in nginx.conf in the http context.  That worked for me.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=blaisorblade">blaisorblade</a>:</i>
<p>I ran into this yesterday. What's the status? Somebody proposed an <tt>X-Jenkins-Crumb</tt> header (in <a href="https://issues.jenkins.io/browse/JENKINS-7518" title="CLONE -Crumb breaks ajax request behind proxies. -- Still broken behind nginx proxies" class="issue-link" data-issue-key="JENKINS-7518"><del>JENKINS-7518</del></a>), which doesn't seem a legal JavaScript identifier (though it might be "the start of a legal JavaScript statement").<br/>
Would <tt>X-Jenkins-Crumb-0NoJS</tt> help, since <tt>0NoJs</tt> is an illegal JS token?</p>

<p>Also, SECURITY-47 isn't visible in the bug tracker, apparently on purpose — I've only found the fix (<a href="https://github.com/jenkinsci/jenkins/commit/0de3e9b14ed75f70279435e78eb9f6a3a1a179df" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/commit/0de3e9b14ed75f70279435e78eb9f6a3a1a179df</a>) and maybe the advisory (<a href="https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2013-02-16" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2013-02-16</a>).</p>

<p>For my part, I tried to document this on <a href="https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+behind+an+NGinX+reverse+proxy" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+behind+an+NGinX+reverse+proxy</a>; I only mentioned the most secure workaround, not the <tt>-Dhudson.security.csrf.requestfield=</tt> setting. Improvements are welcome.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timboudreau">timboudreau</a>:</i>
<p>Running into this too, with NginX, on <a href="https://timboudreau.com/builds" class="external-link" target="_blank" rel="nofollow noopener">https://timboudreau.com/builds</a></p>

<p>I've set ignore_invalid_headers to off, but I'm not keen on leaving it that way, especially for an NginX instance that is proxying far more than just Jenkins - there are other forms of nastiness that would be better not to pass on to back end applications that makes possible.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tomreyn">tomreyn</a>:</i>
<p>The number of people watching this issue is pretty high compared to that of other issues. This may provide an indication as to how much of an issue it is.</p>

<p>C.f. <a href="http://seclists.org/bugtraq/2015/Aug/161" class="external-link" target="_blank" rel="nofollow noopener">http://seclists.org/bugtraq/2015/Aug/161</a></p>

<p>Adding a couple HTTP headers would also help to harden the security of a Jenkins installation (may break some Jenkins plugins - works for me). Assuming your domain name is EXAMPLE.ORG and you use HTTPS:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Content-Security-Policy: "default-src 'self' EXAMPLE.ORG *.EXAMPLE.ORG; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src *; media-src *; object-src 'self'; frame-src 'self' updates.jenkins-ci.org;"
X-Frame-Options: "sameorigin, DENY"
X-XSS-Protection: "1; mode=block"
X-Content-Type-Options:"nosniff"
Strict-Transport-Security:"max-age=15768000"
</pre>
</div></div>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timboudreau">timboudreau</a>:</i>
<p>> as crumb is a valid Javascript identifier that would open you to attack</p>

<p>Would like to understand better how a header name being a Javascript identifier opens one to attack - there are plenty of standard HTTP header names that are valid identifiers - if I make an HTTP request containing</p>

<p>Date: eval("alert('Hey')")</p>

<p>is something foolish enough to execute it?</p>

<p>At any rate, the problem here is simple - ".crumb" is not a legal HTTP header, but there are plenty of options that are legal HTTP headers and illegal as js identifiers - one has already been suggested.  I don't think any of us want to leave our NginX installs set to accept whatever garbage a client decides to turn into headers.</p>

<p>Re the suggested additional headers:  Some are not a bad idea, but I could write a NodeJS proxy that would strip those out in about 10 minutes and go hog wild (or just use a programmatic HTTP client) - you can't rely on that sort of thing to give you real security, just deter low-hanging-fruit bad behavior.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<blockquote><p>Would like to understand better how a header name being a Javascript identifier opens one to attack</p></blockquote>

<p>It is complicated but has to do with how the crumb is offered to legitimate clients.</p>

<p>Anyway the fix is well understood and filed, “just” needs to be tested and reviewed by someone qualified in web security.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 core/src/main/java/hudson/security/csrf/CrumbFilter.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuerDescriptor.java<br/>
 core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java<br/>
 test/src/test/groovy/hudson/model/SlaveTest.groovy<br/>
 test/src/test/java/lib/form/RowSetTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf</a><br/>
Log:<br/>
  Merge pull request #2198 from daniel-beck/<a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a></p>

<p><span class="error">&#91;FIX JENKINS-12875&#93;</span> Change default crumb name to Jenkins-Crumb</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek">jknurek</a>:</i>
<p>Maybe it's worth noting in this ticket:<br/>
I installed Jenkins 2.0-rc-1 and created a view using Build Monitor View plugin (1.8+build.201601112328), and get this error (which leaves the UI in an infinite "sorry" loop</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Apr 14, 2016 8:16:03 AM hudson.security.csrf.CrumbFilter doFilter
WARNING: No valid crumb was included in request <span class="code-keyword">for</span> /$stapler/bound/18964be5-e86d-4242-a57b-35ee573d52c9/fetchJobViews. Returning 403.
</pre>
</div></div>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek">jknurek</a>:</i>
<p>My first thought was that it was a bug specific to the Build Monitor View plugin, but when I run Jenkins 1.651.1, the same configuration works ok. I'm only getting this error with the 2.0-rc-1. <br/>
(but I'm not sure what the underlying issue is in this case, so it is likely a separate issue)</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek">janek</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> Build Monitor has supported CSRF protection since its early days <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>The problem is caused by change <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a> , which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p>

<p>I can work around that in Build Monitor.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> you can watch the progress of this issue on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215</a>, which is the official issue tracker for Build Monitor.</p>

<p>Thanks!<br/>
Jan</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek" class="user-hover" rel="janek">janek</a></p>

<blockquote><p>which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p></blockquote>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">The crumb name has been configurable since 1.310</a> (<tt>hudson.security.csrf.requestfield</tt>), and that was necessary for anyone running Jenkins behind nginx, as it dropped the <tt>.crumb</tt> header (<a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a>). So we renamed the default to something nginx accepts. Maybe it's a good idea to check which crumb name is used by Jenkins in your plugin to adapt to user configuration.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek">janek</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> - Fixed in Build Monitor 1.9+build.2016043001​28, which you can <a href="https://smartcode-opensource.ci.cloudbees.com/job/build-monitor/37/artifact/build-monitor-plugin/target/build-monitor-plugin.hpi" class="external-link" target="_blank" rel="nofollow noopener">download here</a> before it hits the official update centre.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek">janek</a>:</i>
<p>Awesome, thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, that's exactly what I did. The latest Build Monitor picks up the crumb name from CrumbIssuer::getCrumbRequestField so it should be a more appropriate solution <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nvn">nvn</a>:</i>
<p>Hi,<br/>
I have Jenkins 2.7, I have created the service hook in TFS to trigger the automatic build but while testing the hook I get " No valid crumb was included in the request (403)". I can run the same build from Jenkins dashboard with no issue.<br/>
I have also installed the downgrade Build - Plugin-Version: 1.9+build.201606171408 as suggested by Jan Molak above but I am still facing the issue.</p>

<p>P.s - I have installed Jenkins on my local windows 2007 machine.</p>

<p>Thanks,</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek">janek</a>:</i>
<p>Hey Naveen - from what you're saying it seems like the problem is related to Jenkins and not the Build Monitor plugin?<br/>
If you think it's related to Build Monitor please feel free to raise a ticket on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin</a></p>

<p>Thanks,<br/>
Jan</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nvn">nvn</a>:</i>
<p>Hi Jan, yeah I think the issue is either with Jenkins or may be with Tfs-jenkins plugin, I am not sure. Pls suggest.</p>

<p>Thanks</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=12niner">12niner</a>:</i>
<p>Having just wasted a few hours debugging why I was seeing this: if you re-install Jenkins you need to clear your cookies or you'll hit this.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=12niner" class="user-hover" rel="12niner">12niner</a> Too many cookies is tracked as <a href="https://issues.jenkins.io/browse/JENKINS-25046" title="Cookie header too long, causing a 413 HTTP error" class="issue-link" data-issue-key="JENKINS-25046">JENKINS-25046</a>.</p>

**Comment by timja-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-24249">JENKINS-24249</a>]</i>

**Comment by timja-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-27972">JENKINS-27972</a>]</i>

**Comment by timja-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-25116">JENKINS-25116</a>]</i>

**Comment by timja-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-23793">JENKINS-23793</a>]</i>

**Comment by timja-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-7518">JENKINS-7518</a>]</i>

**Comment by timja-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-12787">JENKINS-12787</a>]</i>

**Comment by timja-bot** [comment]
<i>[Epic: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-04/issues?q=in%3Atitle%20JENKINS-33810%20label%3Aepic">JENKINS-33810</a>]</i>
